src/librustc/traits/project.rs

```diff
@@ -1461,7 +1461,7 @@ fn assoc_ty_def(
     // cycle error if the specialization graph is currently being built.
     let impl_node = specialization_graph::Node::Impl(impl_def_id);
     for item in impl_node.items(tcx) {
-        if item.kind == ty::AssocKind::Type
+        if matches!(item.kind, ty::AssocKind::Type | ty::AssocKind::OpaqueTy)
```

Comment:
The fix looks correct to me, but now I'm wondering if the bug! below is reachable for opaque types if the impl is missing the associated opaque type item